xmodule/modulestore/split_mongo/mongo_connection.py

```diff
@@ -437,18 +441,18 @@ def _generate_query_from_course_keys(self, branch, course_keys):
 
         return courses_queries
 
-    def insert_course_index(self, course_index, course_context=None, last_update_already_set=False):
+    def insert_course_index(self, course_index, course_context=None):
```

Comment:
Why is `last_update_already_set` no longer necessary? I mean, I get why it doesn't matter for MySQL, but we don't need to keep using it for the MongoDB-only mode of things? Or is that just not a thing anymore?

Reply:
Oh, we're always deferring to MySQL at this point, right, nvm.

Reply:
The logic here is unchanged, I just moved it to an instance variable (`self.with_mysql_subclass`) instead of passing it as a parameter (`last_update_already_set `) each time this function is called. Because the behavior is either always on or always off, depending on whether the subclass is being used, so it makes sense to set it once during the constructor rather than passing it as a parameter each time.